src/main/java/io/gravitee/policy/oauth2/Oauth2Policy.java

```diff
@@ -139,7 +139,16 @@ public Maybe<SecurityToken> extractSecurityToken(HttpPlainExecutionContext ctx)
 
     @Override
     public Maybe<SecurityToken> extractSecurityToken(KafkaConnectionContext ctx) {
-        return getSecurityTokenFromContext(ctx);
+        return getSecurityTokenFromContext(ctx)
+            .doAfterSuccess(securityToken -> {
+                if (securityToken != null && securityToken.isInvalid()) {
+                    for (Callback callback : ctx.callbacks()) {
+                        if (callback instanceof OAuthBearerValidatorCallback oAuthCallback) {
+                            oAuthCallback.error("invalid_token", null, null);
+                        }
+                    }
+                }
+            });
```

Comment:
Any reason why you used `doAfterSuccess` instead of `doOnSuccess`?
Also, I don't understand why it is made during the token extraction instead of when calling `authenticate(ctx)`